build-llvm.py

```diff
@@ -0,0 +1,402 @@
+                        reporting problems to LLVM developers but will make compilation of both LLVM
+                        and the kernel go slower.
+
+                        """),
+                        action="store_true")
+    parser.add_argument("-i",
+                        "--incremental",
+                        help=textwrap.dedent("""\
+                        By default, the script removes all build artifacts from previous compiles. This
+                        prevents that, allowing for dirty builds and faster compiles.
+
+                        """),
+                        action="store_true")
+    parser.add_argument("-I",
+                        "--install-folder",
+                        help=textwrap.dedent("""\
+                        By default, the script will create a "usr" folder in the same folder as this script
+                        and install the LLVM toolchain there. If you'd like to have it installed somewhere
+                        else, pass it to this parameter. This can either be an absolute or relative path.
+
+                        """),
+                        type=str,
+                        default=os.path.join(root.as_posix(), "usr"))
+    parser.add_argument("-n",
+                        "--no-pull",
+                        help=textwrap.dedent("""\
+                        By default, the script always updates the LLVM repo before building. This prevents
+                        that, which can be helpful during something like bisecting.
+
+                        """),
+                        action="store_true")
+    parser.add_argument("-p",
+                        "--projects",
+                        help=textwrap.dedent("""\
+                        Currently, the script only enables the clang, compiler-rt, and lld folders in LLVM. If
+                        you would like to override this, you can use this parameter and supply a list that is
+                        supported by LLVM_ENABLE_PROJECTS.
+
+                        See step #5 here: https://llvm.org/docs/GettingStarted.html#getting-started-quickly-a-summary
+
+                        Example: -p \"clang;lld;libcxx\"
+
+                        """),
+                        type=str,
+                        default="clang;lld;compiler-rt")
+    parser.add_argument("-t",
+                        "--targets",
+                        help=textwrap.dedent("""\
+                        LLVM is multitargeted by default. Currently, this script only enables the arm32, aarch64,
+                        powerpc, and x86 backends because that's what the Linux kernel is currently concerned with.
+                        If you would like to override this, you can use this parameter and supply a list that is
+                        supported by LLVM_TARGETS_TO_BUILD: https://llvm.org/docs/CMake.html#llvm-specific-variables
+
+                        Example: -t "AArch64;X86"
```

Comment:
Another string I hate.  I'm conflicted because distributions SHOULD NOT be disabling backends, but for local development, I'm just as guilty. :(

Reply:
Given that this is targeting users/developers and not distributions, I think it is okay but I can remove this line if you feel strongly about it.

Reply:
No.